advisories/a64325.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Technical Advisory 64325
+summary: Race condition between long running reads and replica removal
+toc: true
+---
+
+Publication date: May 3, 2021
+
+## Description
+
+Cockroach Labs has discovered a race condition, where a long running read request submitted during replica removal may be evaluated on the removed replica, returning an empty result. Common causes of replica removal are range merges and replica rebalancing in the cluster. When a replica is removed there is a small window, where the data is already deleted but a read request will still see it as a valid replica. In this case a read request will always return an empty result for the part of the query that relied on this range.
+
+This is a very rare scenario that can only be encountered, if a number of very unlikely conditions are met:
+
+ - A read request checks that the node is holding the lease and begins processing
+ - The node loses the lease before the read request is able to lock the replica for reading
+ - The read request checks the state of the replica and makes sure it's a alive
+ - The new leaseholder immediately removes the old replica from the old leaseholder, due to a rebalance or range merge.
```

Comment:
Consider re-wording to "the old leaseholder replica from the range".

Also, consider adding "The new leaseholder _decides to_" to hint that this is another independent condition that needs to hold.